src/lib/components/BlogPage.svelte

```diff
@@ -0,0 +1,159 @@
+<script lang="ts">
+  import { createEventDispatcher } from "svelte";
+  import Button from "./Tag.svelte";
+  import BlogLink from "$lib/components/BlogLink.svelte";
+  import type { Tags } from "$lib/utils/types/microcms";
+
+  export let more: boolean = false;
+  export let blogs;
+  export let limit: number = 10;
+
+  let blogListContents = blogs.blogList.contents;
+
+  let sortKey: boolean = false;
+  let filterTag = "All";
+
+  let page = 1;
+
+  $: start = (page - 1) * limit;
+  $: end = start + limit;
+
+  const dispatch = createEventDispatcher();
+
+  function handlePublishedAtClick() {
+    sortKey = !sortKey;
+    dispatch("click");
+  }
+
+  $: sortedAndFilteredBlogs = blogListContents
+    .filter(
+      (blog: { tags: string | string[] }) =>
+        filterTag === "All" || blog.tags.includes(filterTag),
+    )
+    .sort((a: { publishedAt: string }, b: { publishedAt: string }) => {
+      switch (sortKey) {
+        case true:
+          return a.publishedAt.localeCompare(b.publishedAt);
+        case false:
+          return b.publishedAt.localeCompare(a.publishedAt);
+        default:
+          return 0;
+      }
+    })
+    .slice(start, end);
+
+  $: morePage = blogListContents.length > end;
+
+  const tags: Tags[] = [
+    "お知らせ",
+    "ブログ",
+    "作品紹介",
+    "活動報告",
+    "1 年生",
+    "2 年生",
+    "3 年生",
+    "_migrated",
+  ];
+</script>
+
+{#if more}
+  <div id="sort">
+    <div>
+      <select  bind:value={filterTag}>
+        <option value="All">All</option>
+        {#each tags as tag}
+          <option value={tag}>{tag}</option>
+        {/each}
+      </select>
+    </div>
+    <div>
+      <Button on:click={handlePublishedAtClick}>投稿日</Button>
```

Comment:
```suggestion
      <Button on:click={handlePublishedAtClick}>公開日</Button>
```
PublishedAt を訳すなら “公開日” のほうが近いかもですね.  実際, BlogLink のアイキャッチ画像の下にある日付は “公開” になっていますね.